src/tic-tac-toe.ts

```diff
@@ -40,35 +43,73 @@ class Game {
     return square === this.currentPlayer;
   };
 
-  private checkRowsForWin(): boolean {
-    return this.board.some(row => row.every(this.isCurrentPlayer));
+  private checkRowsForWin(): WinningVector | undefined {
+    for (let rowIndex = 0; rowIndex < this.board.length; rowIndex++) {
```

Comment:
You're using `this.board.length`, `this.boardSize`, and `transposedBoard.length` interchangeably in these methods; I would try to be consistent and just stick to one (probably `this.boardSize`).